File: Javascript/75/1004.js
// 1004. Max Consecutive Ones III

var longestOnes = function (nums, k) { 

  let left = 0; 
  let maxLength = 0;
  let zeroCount = 0; 

  for (let right = 0; right < nums.length; right++) {
      // Expand the window
      if (nums[right] === 0) {
          zeroCount++;
    }

      // Shrink the window if the number of 0's exceeds k
    while (zeroCount > k) {
      console.log(zeroCount);
          if (nums[left] === 0) {
              zeroCount--;
          }
          left++;
      }

    // Update the maximum length
      maxLength = Math.max(maxLength, right - left + 1);
  }

  return maxLength;

};



const nums = [1, 1, 1, 0, 0, 0, 1, 1, 1, 1, 0],
  k = 2;
    
  console.log(longestOnes(nums, k));

File: Leetcode/1004.py
from typing import List


class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        left = 0
        max_length = 0
        zero_count = 0

        for right in range(len(nums)):
            # Expand the window
            if nums[right] == 0:
                zero_count += 1

            # Shrink the window if zero count exceeds k
            while zero_count > k:
                if nums[left] == 0:
                    zero_count -= 1
                left += 1

            # Update maximum length
            max_length = max(max_length, right - left + 1)

        return max_length


# Examples
solution = Solution()

# Example 1
nums1 = [1, 1, 1, 0, 0, 0, 1, 1, 1, 1, 0]
k1 = 2
print(solution.longestOnes(nums1, k1))  # Output: 6

# Example 2
nums2 = [0, 0, 1, 1, 0, 0, 1, 1, 1, 0, 1, 1, 0, 0, 0, 1, 1, 1, 1]
k2 = 3
print(solution.longestOnes(nums2, k2))  # Output: 10
